nixos/modules/services/networking/whoogle-search.nix

```
@@ -0,0 +1,124 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.whoogle-search;
+  whoogle-search = pkgs.whoogle-search;
+  stateDir = "/var/lib/whoogle-search";
+in
+{
+  options = {
+    services.whoogle-search = {
+
+      enable = mkEnableOption "Whether to enable whoogle-search service.";
+
+      user = mkOption {
+        type = types.str;
+        default = "whoogle-search";
+        description = "The user as which to run whoogle-search daemon.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "whoogle-search";
+        description = "The group as which to run whoogle-search daemon.";
+      };
+
+      port = mkOption {
+        type = types.port;
+        default = 5000;
+        description = ''
+          Port to listen on.
+        '';
+      };
+
+      listenAddress = mkOption {
+        type = types.str;
+        default = "127.0.0.1";
+        description = ''
+          Address to listen on for the web interface.
+        '';
+      };
+
+      dataDir = mkOption {
+        type = types.path;
+        default = stateDir;
+        description = ''
+          The directory where whoogle-search will create session files.
+        '';
```

```suggestion
        description = ''
          The directory where whoogle-search will create session files.
          If left as the default value this directory will automatically be created before whoogle-search starts, otherwise the sysadmin is responsible for ensuring the directory exists with appropriate ownership and permissions.
        '';
```

See my comment below in the `preStart` script for explanation.